src/main/java/co/refiere/dao/PersonDao.java

```diff
@@ -40,11 +33,15 @@ public StatelessSession getStatelessSession(){
     public void save(Person person){
         Session session = sessionFactory.getCurrentSession();
         org.hibernate.Transaction trans= session.beginTransaction();
-        merge(person);
+        if(person.getId() == 0){
```

Comment:
this should stay as persist(person);
Why are you using differents methods?
In that case, I think you probably want to use: session.saveOrUpdate(person); which saves or update a object instead of adding a if-else statement.